README.md

```diff
@@ -14,62 +14,33 @@ certification.  Please see "CNF Developers" below for more information.
  ![overview](docs/images/overview.svg)
 
 In the diagram above:
-- the `CNF under test` is the CNF to be certified. The certification suite identifies the pods belonging to the CNF via a label, see [Pod Roles](#pod_roles)
+- the `CNF under test` is the CNF to be certified. The certification suite identifies the resources (containers/pods/operators etc) belonging to the CNF via labels or static data entries in the config file
 - the `Certification container/exec` is the certification test suite running on the platform or in a container. The executable verifies the CNF under test configuration and its interactions with openshift 
-- the `Partner pod` can be any pod with the required tools (e.g. only ping at the moment) in the same namespace as the `CNF under test`. During connectivity tests, the partner pod will generate pings towards the `CNF under test` to verify connectivity.
-
-These roles are configured in the [Test Configuration](#test-configuration) section below.
+- the `Partner pod` can be any pod with the required tools in the same namespace as the `CNF under test`. For example, during connectivity tests, the partner pod will generate pings towards the `CNF under test` to verify connectivity. The partner pods/containers are auto deployed by the test suite prior a test run and can be auto discovered by the suite without any data entry in the config file.
 
 
 ## Test Configuration
 
-Detailed configuration of the individual specs is explained in [config.md](docs/config.md).
-
-By leveraging resource labels. Automatic configuration will happen by default, that being said, it can be disabled if the environment variable `TNF_DISABLE_CONFIG_AUTODISCOVER` is set
-
-```shell
-TNF_DISABLE_CONFIG_AUTODISCOVER=true
-```
-
-Pods can be labelled at creation by including the label in their definition, or at any time using the `oc label`
-command. Annotations are almost identical:
+A sample config file can be found [here](test-network-function/tnf_config.yml). It consists of the following sections:
 
-```shell
-oc label pod test -n tnf test-network-function.com/generic=target
-oc annotate pod test -n tnf test-network-function.com/skip_connectivity_tests=true
-```
+### targetPodLabels
 
+This section contains a list of labels that will be used to discover the pods and their containers to be tested **in addition to** the ones
+explicitly configured in the testTarget sections.
+### testTarget
+#### podsUnderTest / containersUnderTest
```

Comment:
### podsUnderTest / containersUnderTest

##### Configuration via config file
See below an commented example of adding a single containerand a single pod to test:

```shell script
testTarget:
  containersUnderTest:
    - namespace: tnf # the namespace where the test are run
      podName: test # the name of the pod under test 
      containerName: test the name of the container under test 
      defaultNetworkDevice: eth0 # the name of the default network device (the one assigned by kubernetes core mechanism, not multus)
      multusIpAddresses: # the multus asigned IP
        - 10.217.0.8
  podsUnderTest:
    - name: test # The name of the pod under test
      namespace: tnf # the namespace where the test are run
      tests: # List of tests to run for this pod (List TBD)
        - PRIVILEGED_POD
        - PRIVILEGED_ROLE
```


##### Auto-discovery with labels
Label the pods under test with the target pod label specified in the [targetpodlabels](targetPodLabels) section. 
It's highly recommended that the labels shoud be defined in pod definition rather than added after pod is created, as labels added later on will be lost in case the pod gets rescheduled. In case of pods defined as part a deployment, it's best to use the same label as the one defined in the `spec.selector.matchLabels` section of the deployment yaml.

The autodiscovery mechanism will also attempt to identify the default network device and all the IP addresses of the pods it
needs for network connectivity tests, though that information can be explicitly set using annotations if needed. For Pod IPs: